cmd/egress-cni-plugin/snat/snat.go

```diff
@@ -91,31 +92,45 @@ func Add(ipt iptableswrapper.IPTablesIface, nodeIP, src net.IP, multicastRange,
 }
 
 // Del removes rules added by snat
-func Del(ipt iptableswrapper.IPTablesIface, src net.IP, chain, comment string) error {
-	err := ipt.Delete("nat", "POSTROUTING", "-s", src.String(), "-j", chain, "-m", "comment", "--comment", comment)
-	if err != nil && !isNotExist(err) {
-		return err
+func Del(ipt iptableswrapper.IPTablesIface, src net.IP, chain, comment string) (err error) {
+	var ipAddr string
+	if src == nil {
+		// since src ip is not provided, we list all nat POSTROUTING rules then filter by matching chain
+		// which is a hash of container id as part of its value, and it is unique for each container's nat POSTROUTING rule
+		// if no matching rules found, it means rule had already been removed
+		var rules []string
+		rules, err = ipt.List("nat", "POSTROUTING")
+		for _, rule := range rules {
```

Comment:
Looping through all of the rules can be an expensive operation, depending on the size of the cluster. I think we should stick with hard-coding the deletes

Reply:
I did more test in eks cluster, and found out that `DEL` action was called 3 times in a normal pod removal, the first time with netns and preResult having value, the following 2 without preResult and netns value.

The currently production plugins worked the same.

Based on above observation, I changed the `del` code back to original version -- without loop through links to find src ip.